auth-service/src/main/kotlin/uk/gov/communities/delta/auth/controllers/internal/EditAccessGroupsController.kt

```diff
@@ -59,12 +123,67 @@ class EditAccessGroupsController(
 
         val accessGroupActions = generateAccessGroupActions(accessGroupRequestMap, currentAccessGroups, selectedOrgs)
 
-        executeAccessGroupActions(accessGroupActions, session, callingUser, call)
+        executeAccessGroupActions(accessGroupActions, null, callingUser, call)
 
         return call.respond(mapOf("message" to "Access groups have been updated. Any changes to your roles or access groups will take effect the next time you log in."))
     }
 
-    private suspend fun validateRequest(
+    fun validateAddAccessGroupRequest(
+        groupName: String,
+        callingUser: LdapUser,
+        targetUser: LdapUser,
+        allAccessGroupNames: List<String>,
+    ) {
+        validateGroupExistenceAndUserAuthority(allAccessGroupNames, groupName, callingUser)
+
+        if (targetUser.memberOfCNs.contains(LDAPConfig.DATAMART_DELTA_PREFIX + groupName)) {
+            throw ApiError(
+                HttpStatusCode.Forbidden,
```

Comment:
Nit most of these should be 400 (or maybe 409 or 422), providing different credentials is never going to make this request succeed.